shopinvader_auth_jwt/models/shopinvader_backend.py

```diff
@@ -18,12 +18,30 @@ class ShopinvaderBackend(models.Model):
 
     @api.model
     def _get_from_jwt_aud(self, aud):
+        """On a single record, ensure aud is valid
```

Comment:
Hm, that is really confusing. Since this method is declared as `api.model`, it should not work on a record set.
Better make two methods, then.

Reply:
Btw, i'm not sure if we really need to support the use case with the empty recordset.

Reply:
> Btw, i'm not sure if we really need to support the use case with the empty recordset.

Probably not

Originally this method was simply a class method to get the backend matching an aud claim.
But then the code evolved because we wanted to check that the backend selected with the website key also matches the jwt aud... and it became messy and buggy.

So yes, more cleanup needed, I'm afraid.

Reply:
Cleanup done :)